pymatgen/core/structure.py

```diff
@@ -49,6 +49,7 @@
 if TYPE_CHECKING:
     from collections.abc import Iterable, Iterator, Sequence
     from pathlib import Path
+    from typing import Self
```

Comment:
A new trick for me. But I get:
```python
pymatgen/core/structure.py:52: error: Module "typing" has no attribute "Self"  [attr-defined]
pymatgen/core/structure.py:52: note: Use `from typing_extensions import Self` instead
pymatgen/core/structure.py:52: note: See https://mypy.readthedocs.io/en/stable/runtime_troubles.html#using-new-additions-to-the-typing-module
```

I might replace these. @janosh

Reply:
feel free to remove. i forgot to remove myself. it's only supported on 3.11+ but i was curious to check if it would solve the `mypy` error (it didn't)

Reply:
Look like you are using it in the absolutely correct form, just replacing it with `from typing_extensions import Self` is sufficient to fix this issue below Python 3.11.